### PR TITLE
Loan form Expire date input added

### LINKED ISSUE #365: Add expire date input to loan form
## ✨ Feature request
Add expire date input field

### 📇 User story
As a Admin i want to be able to set the expire date

### 📜 Acceptance Criteria
- [ ] Input field added

### PR DESCRIPTION
**Pull request by TE4-Sebastian-Utbult**
## ⚙️ What issue did you implement or fix?
Fixes #365

## 📜 Summary
- Fixed foobar in zoobar.
- Made foo better.

## 📝 Type of change

Please delete options that are not relevant.

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 This change requires a documentation update

## ✅ Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## 💩 How can this implementation be improved?
- Change bar into zar. :shipit:

### CODE REVIEW

**Comment by te4-adrian-almetun-smeds** [comment]
Try running `git reset --soft HEAD~2` and then do `git add frontend/cypress/integration/loanBooks.js` `frontend/src/components/LoanFormComponent.vue` `frontend/src/types.ts`

**Comment by te4-david-jensen** [comment]
This branch is (as of writing) 206 commits behind 'develop'. I believe you'll need to get it up to date in order for these checks to pass.

File: frontend/src/components/LoanFormComponent.vue [hunk 1]
```diff
@@ -62,21 +95,23 @@ import { ref, defineComponent, SetupContext} from '@vue/composition-api';
...
 // authorized by the pundit dependency 
+
 export default defineComponent({
   name: 'LoanFormComponent',
   setup(_ : object, {root, emit} : SetupContext) {
     const form = {
       lent_by_id: 0, //eslint-disable-line camelcase
       loaned_by_id: 0, //eslint-disable-line camelcase
-      book_id: 0 //eslint-disable-line camelcase
+      book_id: 0,
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/LoanFormComponent.vue:105]:
```suggestion
      book_id: 0, //eslint-disable-line camelcase
```

File: frontend/src/types.ts [hunk 2]
```diff
@@ -107,6 +107,7 @@ interface LoanForm {
   book_id: number; //eslint-disable-line camelcase
   lent_by_id: number; //eslint-disable-line camelcase
   loaned_by_id: number; //eslint-disable-line camelcase
+  expiration_date: string;
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/types.ts:110]:
```suggestion
  expiration_date: string; //eslint-disable-line camelcase
```

**Review by te4-adrian-almetun-smeds — CHANGES_REQUESTED** [review]
Some lint errors but, should be a quick fix

File: frontend/src/components/LoanFormComponent.vue [hunk 3]
```diff
@@ -34,6 +35,38 @@
...
+              outlined
+              v-bind="attrs"
+              v-on="on"
+            />
+          </template>
+          <v-date-picker
+            ref="picker"
+            v-model="form.expiration_date"
+            :min="new Date().toISOString().substr(0, 10)"
+            @change="save"
```
Comment (te4-adrian-almetun-smeds) [inline frontend/src/components/LoanFormComponent.vue:65]: Save doesn't seem to be defined in this component. Should it be?

**Review by TE4-Rasmus-Brednert — CHANGES_REQUESTED** [review]
Look at the failed checks, then you're ready to go.

**Review by te4-andre-skvarc — CHANGES_REQUESTED** [review]
Looks better now, but check the test so they pass. I recommend doing a rebase and then push up the code again.

**Review by te4-andre-skvarc — APPROVED** [review]
Looks good now!  :shipit:

**Comment by itggot-linus-styren** [comment]
Why are there two almost identical methods with different types:
https://github.com/itggot-TE4/Yabs/blob/51765cf64d7dc17a2e726d6cc67a76b0d3936f65/frontend/src/services/api/loans.ts#L14-L28

The first one is using LoanForm and the other one is using AddLoanForm?

**Review by TE4-Rasmus-Brednert — APPROVED** [review]
Nice!

**Comment by te4-adrian-almetun-smeds** [comment]
> Why are there two almost identical methods with different types:
> https://github.com/itggot-TE4/Yabs/blob/51765cf64d7dc17a2e726d6cc67a76b0d3936f65/frontend/src/services/api/loans.ts#L14-L28
> 
> The first one is using LoanForm and the other one is using AddLoanForm?
> 
> Maybe the second one can be removed to resolve codeclimate complaint. It doesn't seem to be used anywhere:
> image

Good question. A new issue has been open regarding said issue since it's outside of the scope of this PR.

**Review by te4-adrian-almetun-smeds — APPROVED** [review]

**Pull request merged**